crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -0,0 +1,599 @@
+//! Types for creating and storing [`Observer`]s
+
+mod builder;
+mod entity_observer;
+mod runner;
+
+use std::marker::PhantomData;
+
+pub use builder::*;
+pub(crate) use entity_observer::*;
+pub use runner::*;
+
+use crate::{
+    archetype::ArchetypeFlags,
+    query::DebugCheckedUnwrap,
+    system::{EmitEcsEvent, IntoObserverSystem},
+    world::*,
+};
+
+use bevy_ptr::{Ptr, PtrMut};
+use bevy_utils::{EntityHashMap, HashMap};
+
+use crate::{component::ComponentId, prelude::*, world::DeferredWorld};
+
+/// Type used in callbacks registered for observers.
```

Comment:
These docs need to be much more approachable: explaining the purpose of observers and how you might use them with a doc test.

Reply:
I think that probably belongs on the Observer type now that the rename has happened. I'll expand the docs here though